### ISSUE TITLE
CategoricalPd:entropy

### ISSUE THREAD

**Issue by Wang4758** [openai/baselines#1086]
I have some trouble understanding the calculation of entropy for discrete action space (CategoricalPd). 
```    
def entropy(self):
        a0 = self.logits - tf.reduce_max(self.logits, axis=-1, keepdims=True)
        ea0 = tf.exp(a0)
        z0 = tf.reduce_sum(ea0, axis=-1, keepdims=True)
        p0 = ea0 / z0
        return tf.reduce_sum(p0 * (tf.log(z0) - a0), axis=-1) 
```

Could you please give me some explanation for this? Thanks a lot!

**Comment by GNiendorf** [comment]
First four lines are just calculating the softmax probabilities from the logits. the line that returns the entropy is just -p0log(p0) = -p0log(ea0/z0) = p0log(z0/ea0) = p0 * (log(z0) - log(ea0)) = p0 * (log(z0) - a0)
